src/framework/xr/xr-manager.js

```diff
@@ -25,6 +25,13 @@ import { XrAnchors } from './xr-anchors.js';
  * @param {Error|null} err - The Error object or null if operation was successful.
  */
 
+/**
+ * Callback used by manual room capturing.
+ *
+ * @callback XrRoomCapture
```

Comment:
I think we usually add `Callback` like `XrRoomCaptureCallback`, but I started to wonder, do we still want to add more callbacks everywhere when Promises are becoming the new standard to prevent callback hell?

Reply:
Well spotted.
Yes, we want a **consistent API**, so we use callbacks as it is a common practice around the engine.
Promises do not prevent callback hell, async/await do, but they are not always desirable.

Refactoring / adding promise alternatives to callback-like methods should be done as a separate PR if users will demand a need for and report that there is a callback hell (so far API is clean and devs have not highlighted it as an issue).